educator/src/Dscp/Educator/Web/Auth.hs

```diff
@@ -122,13 +264,48 @@ checkAuthBasic = do
     checkAuthData authData
     pure pk
 
+-- | Sign authentiation data and produce JWT token.
+-- Second argument stands for endpoint name, example:
+-- @/api/educator/v1/students@.
+authDataToJWT :: SecretKey -> AuthData -> ByteString
+authDataToJWT sk = LBS.toStrict . encodeCompact . signJWitness sk . encode
+
+-- | Create an authentication token.
+createAuthToken :: MonadIO m => SecretKey -> Text -> m ByteString
+createAuthToken secretKey endpoint = do
+    time <- liftIO getCurrentTime
+    let authData = AuthData
+            { adPath = endpoint
+            , adTime = time
+            }
+    return $ authDataToJWT secretKey authData
+
+-- | Make a test authentication token.
+-- You can pass this to @curl@ as @-H "Authorization: Bearer <produced text>"@.
+makeTestAuthToken :: SecretKey -> Text -> ByteString
+makeTestAuthToken secretKey endpoint = authDataToJWT secretKey authData
+  where
+    farFuture = posixSecondsToUTCTime 1735689600 -- 1 Jan 2025
+    authData = AuthData { adPath = endpoint, adTime = farFuture }
+
+-- | Extract endpoint name from built request.
+clientRequestEndpoint :: Cli.Request -> Text
+clientRequestEndpoint = decodeUtf8 . toLazyByteString . Cli.requestPath
+
+-- | Make authentication signature.
+signRequestBasic :: SecretKey -> Cli.Request -> AuthToken
+signRequestBasic sk req =
+    let endpoint = decodeUtf8 . toLazyByteString $ Cli.requestPath req
+    in AuthToken $ "Bearer " <> makeTestAuthToken sk endpoint
+        ---- ^ TODO: provide not a test auth token, but that requires IO :(
```

Comment:
So what?...

Reply:
Right. Since this is going to be used in production, I will add some `unsafePerformIO`s.

Reply:
@Martoon-00 well, at first you can make `provideAuth` monadic and see how it goes. This would be a valid change anyway, because construction of auth token often requires something from the outside world (timestamp, random, whatever)

Reply:
Yeah, I want to try to propagate this `IO` addition to `instance HasServer`, maybe I won't even need unsafe things at all.